### ISSUE TITLE
Compile-on-build conflicting with Compile-on-save

### ISSUE THREAD

**Issue by jeradrose** [Microsoft/TypeScript#5424]
It seems that when Visual Studio compiles TS into JS, there are some inconsistencies, resulting in a ping-pong match with commits. Specifically, when compiling from a solution build, the yielded .js is different from the .js resulting from compiling from a save.

For one (simple) example, compile-on-save will use enums:

    _this.kernelState(NotebookKernelState.Queued);

Where compile-on-build will use constants w/ comments:

    _this.kernelState(5 /* TimeExpired */);

Another example is how they vary their formats (this may be related to ReSharper settings, though not sure how to make these sync between build compiles and save compiles):

Compile-on-save:

    catch (e) { }

Compile-on-build:

    catch (e) {
    }

I'll also see the `sourceMappingUrl` comment get removed with compile-on-save:

    //# sourceMappingURL=myfile.js.map

Then get added back on with compile-on-build.

Is this a bug, or is it something that we may have misconfigured somewhere?

**Comment by mhegazy** [comment]
Which version of TS and VS are you using? and can you share a repro project?

**Comment by jeradrose** [comment]
The attached project (remove `.png`) can repro the `.map` comment and the `enum`. Note that the enum difference only appears when targeting 1.4 -- targeting 1.6 makes these consistent between save and build compiles. But the `.map` comment is there in 1.4 and 1.6.

I couldn't repro the formatting difference in this project (yet), so I'm thinking that has something to do with project-level ReSharper settings. If I can get this to repro, I'll let you know.

![reprotypescriptcompile zip](https://cloud.githubusercontent.com/assets/743963/10776652/a57fd36e-7ce9-11e5-8f7c-ef4190d35ed8.png)

**Comment by mhegazy** [comment]
Thanks for the repro. the first issue has been fixed in TS 1.5 (see #1812). the formatting issue is strange, have not seen this before. as for the last issue, looks like this project file does not have any configuration options. adding config options, either manually by unloading the project or by saving from the project property pages solves the issue:
```xml
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
    <TypeScriptSourceMap>True</TypeScriptSourceMap>
  </PropertyGroup>
```

how was this project created? is that from template? or hand authored?

**Comment by jeradrose** [comment]
It was created using VS empty web project template.

I will try the source map config to see if that fixes that issue.

**Comment by mhegazy** [comment]
looks like we are not loading the property correctly. @paulvanbrenk can you take a look. i see the source map checkbox checked correctly in the project property pages but not passed correctly to the compiler..

**Comment by paulvanbrenk** [comment]
This looks like it's fixed in 1.8
